packages/web3-core-method/src/index.js

```diff
@@ -770,19 +771,15 @@ Method.prototype.buildCall = function () {
         };
 
         // Send the actual transaction
-        if (isSendTx && !!payload.params[0] && typeof payload.params[0] === 'object' && typeof payload.params[0].gasPrice === 'undefined') {
+        if (isSendTx 
+            && !!payload.params[0]
+            && typeof payload.params[0] === 'object'
```

Comment:
if we remove `typeof payload.params[0].gasPrice === 'undefined'` control never goes to https://github.com/ChainSafe/web3.js/blob/0299df8142006aa1cce28b567a42711ef2acb4bb/packages/web3-core-method/src/index.js#L794  for ganache tests and it fails. 

Plus I think if gas fileds are already with incoming object we can procede to `sendRequest()` without need of moving inside https://github.com/ChainSafe/web3.js/blob/0299df8142006aa1cce28b567a42711ef2acb4bb/packages/web3-core-method/src/index.js#L777 block.

so I suggest 
```suggestion
            && typeof payload.params[0] === 'object' 
            && typeof payload.params[0].gasPrice === 'undefined' 
            && typeof payload.params[0].maxPriorityFeePerGas === 'undefined'
            && typeof payload.params[0].maxFeePerGas === 'undefined'
```

Reply:
Made a modified version of you suggestion, since the user could not provide `gasPrice` or `maxPriorityFeePerGas` and `maxFeePerGas` (the latter two both be required if one is present)